podcasts/EpisodesDataManager.swift

```diff
@@ -3,6 +3,30 @@ import PocketCastsServer
 import DifferenceKit
 
 class EpisodesDataManager {
+    // MARK: - Playlist episodes
+
+    /// Return the list of episodes for a given playlist
+    func episodes(for playlist: AutoplayHelper.Playlist) -> [BaseEpisode] {
+        switch playlist {
+        case .podcast(uuid: let uuid):
+            if let podcast = DataManager.sharedManager.findPodcast(uuid: uuid, includeUnsubscribed: true) {
+                return Array(episodes(for: podcast).compactMap { $0.elements as? [ListEpisode] }.joined()).map { $0.episode }
```

Comment:
I ran into an issue where the autoplay wouldn't work when played from the podcast details when my podcast is grouped by season and sorted newest to oldest. 

After some digging it looks like the data structure returned is:

```swift
[
    ArraySection("Search", [ListHeader()]),
    ArraySection("episodes", [
        ListHeader("No Season"),
        ListEpisode()
        ListEpisode(),
        .....
        ]
    )
]
```

This means that the first compactMap will return nil for the episodes section because not all items are ListEpisodes. 

We can fix this and refactor the code to be a bit faster by using a flatMap to flatten the array, and validating each element of the array.

```suggestion
                return episodes(for: podcast).flatMap { $0.elements.compactMap { ($0 as? ListEpisode)?.episode } }
```